yt_dlp/extractor/kenh14.py

```diff
@@ -0,0 +1,150 @@
+            'formats': [
+                {'url': f'https://{direct_url}', 'format_id': 'http'},
+                *self._extract_m3u8_formats(f'https://{direct_url}/master.m3u8', video_id, fatal=False),
+            ],
+            'duration': parse_duration(metadata.get('duration')),
+            'description': (
+                clean_html(self._og_search_description(webpage))
+                or clean_html(get_element_by_class('vdbw-sapo', webpage))),
+            'thumbnail': (self._og_search_thumbnail(webpage) or attrs.get('data-thumb')),
+            'uploader': strip_or_none(metadata.get('author')),
+            'timestamp': parse_iso8601(metadata.get('uploadtime'), delimiter=' '),
+            'view_count': int_or_none(metadata.get('views')),
+            'tags': traverse_obj(self._html_search_meta('keywords', webpage), (
+                {lambda x: x.split(';')}, lambda _, v: v, {str_or_none})),
+        }
+
+
+class Kenh14PlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://video\.kenh14\.vn/playlist/[\w-]+-(?P<id>[0-9]+)\.chn'
+    _TESTS = [{
+        'url': 'https://video.kenh14.vn/playlist/tran-tinh-naked-love-mua-2-71.chn',
+        'info_dict': {
+            'id': '71',
+            'title': 'Trần Tình (Naked love) mùa 2',
+            'description': 'md5:e9522339304956dea931722dd72eddb2',
+            'thumbnail': r're:^https?://kenh14cdn\.com/.*\.png$',
+        },
+        'playlist_count': 9,
+    }, {
+        'url': 'https://video.kenh14.vn/playlist/0-72.chn',
+        'info_dict': {
+            'id': '72',
+            'title': 'Lau Lại Đầu Từ',
+            'description': 'Cùng xem xưa và nay có gì khác biệt nhé!',
+            'thumbnail': r're:^https?://kenh14cdn\.com/.*\.png$',
+        },
+        'playlist_count': 6,
+    }]
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        webpage = self._download_webpage(url, playlist_id)
+
+        category_detail = get_element_by_class('category-detail', webpage)
+        embed_info = traverse_obj(
+            self._yield_json_ld(webpage, playlist_id),
+            (lambda _, v: v['name'] and v['alternateName']), get_all=False)
+
+        return self.playlist_from_matches(
+            get_elements_html_by_class('video-item', webpage), playlist_id,
+            (clean_html(get_element_by_class('name', category_detail)) or unescapeHTML(embed_info.get('name'))),
+            getter=lambda x: 'https://video.kenh14.vn/video/{}.chn'.format(extract_attributes(x)['data-id']),
+            ie=Kenh14VideoIE.ie_key(), playlist_description=(
+                clean_html(get_element_by_class('description', category_detail))
+                or unescapeHTML(embed_info.get('alternateName'))),
+            thumbnail=traverse_obj(
+                self._og_search_thumbnail(webpage),
+                ({url_or_none}, {lambda x: urllib.parse.urlunparse(urllib.parse.urlparse(x)._replace(query=None))})))
```

Comment:
merge master again, import `update_url` from utils:
```suggestion
                ({url_or_none}, {(update_url(query=None)})))
```